Tests/CatalogItems/New-RsDataSource.Tests.ps1

```diff
@@ -0,0 +1,184 @@
+        $dataSourceName = 'SutDataSurceStoreException' + [guid]::NewGuid()
+        $extension = Get-ExistingExtension
+        $credentialRetrieval = 'STORE'
+        $dataSourcePath = '/' + $dataSourceName
+        It "Should throw an exception when Store credential retrieval are given without providing credential" {
+            { New-RsDataSource -RsFolder '/' -Name $dataSourceName -Extension $extension -CredentialRetrieval $credentialRetrieval } | Should throw 
+            { Get-RsDataSource -Path $dataSourcePath } | Should throw
+        }
+    }
+
+    Context "Create RsDataSource with Data Source Credentials" {
+        # Declare datasource Name, Extension, CredentialRetrieval, and DataSource path.
+        $dataSourceName = 'SutDataSurceCredentials' + [guid]::NewGuid()
+        $extension = Get-ExistingExtension
+        $credentialRetrieval = 'None'
+        $dataSourcePath = '/' + $dataSourceName
+        # Creation of PSCredentials
+        $password ='MyPassword'
+        $user = [System.Security.Principal.WindowsIdentity]::GetCurrent().Name
+        $dataSourceCredentials = Create-PSCredential -User $user -Password $password
+        New-RsDataSource -RsFolder '/' -Name $dataSourceName -Extension $extension -CredentialRetrieval $credentialRetrieval -DataSourceCredential $dataSourceCredentials
+        It "Should be a new data source" {
+            {Get-RsDataSource -Path $dataSourcePath } | Should not throw
+        }
+        # Removing folders used for testing
+        Remove-RsCatalogItem -RsFolder $dataSourcePath
+    }
+
+    Context "Create RsDataSource with ImpersonateUser Parameter"{
+        # Declare datasource Name, Extension, CredentialRetrieval, and DataSource path.
+        $dataSourceName = 'SutDataSurceImpersonateUser' + [guid]::NewGuid()
+        $extension = Get-ExistingExtension
+        $credentialRetrieval = 'None'
+        $dataSourcePath = '/' + $dataSourceName
+        New-RsDataSource -RsFolder '/' -Name $dataSourceName -Extension $extension -CredentialRetrieval $credentialRetrieval -ImpersonateUser
+        It "Should be a new data source" {
+            {Get-RsDataSource -Path $dataSourcePath } | Should not throw
+        }
+        # Removing folders used for testing
+        Remove-RsCatalogItem -RsFolder $dataSourcePath
+    }
+
+    Context "Create RsDataSource with Windows Credentials Parameter"{
+        # Declare datasource Name, Extension, CredentialRetrieval, and DataSource path.
+        $dataSourceName = 'SutDataSurceWindowsCredentials' + [guid]::NewGuid()
+        $extension = Get-ExistingExtension
+        $credentialRetrieval = 'None'
+        $dataSourcePath = '/' + $dataSourceName
+        # Test if the DataSource is created
+        New-RsDataSource -RsFolder '/' -Name $dataSourceName -Extension $extension -CredentialRetrieval $credentialRetrieval -WindowsCredentials
+        It "Should be a new data source" {
+            {Get-RsDataSource -Path $dataSourcePath } | Should not throw
```

Comment:
Could you see if there is any property (after you get the data source) that indicates that Windows Credential will be used when connecting to it?